docs/rfcs/rfc-034-aggregation-query-filters.md

```diff
@@ -0,0 +1,258 @@
+  AND: [PostWhereNoNested!]
+  title: String
+  title_NOT: String
+  title_IN: [String!]
+  title_NOT_IN: [String!]
+  title_CONTAINS: String
+  title_NOT_CONTAINS: String
+  title_STARTS_WITH: String
+  title_NOT_STARTS_WITH: String
+  title_ENDS_WITH: String
+  title_NOT_ENDS_WITH: String
+  released: Int
+  released_NOT: Int
+  released_IN: [Int]
+  released_NOT_IN: [Int]
+  released_LT: Int
+  released_LTE: Int
+  released_GT: Int
+  released_GTE: Int
+}
+
+input GroupPostsAggregateFilter {
+    queryWhere: PostWhereNoNested ## This filter is applied only to the aggregation query
+    aggregationWhere: GroupPostsAggregateInput
+}
+
+input GroupPostsAggregateInput {
+  AND: [GroupPostsAggregateInput!]
+  OR: [GroupPostsAggregateInput!]
+  count: Int
+  count_LT: Int
+  count_LTE: Int
+  count_GT: Int
+  count_GTE: Int
+  node: GroupPostsNodeAggregationWhereInput
+}
+```
+
+To reduce the complexity of the implementation the `{type}Where` type is not being reused during the aggregation filter, as it would have allowed nested filters and/or aggregation query filtered by a nested aggregation query.
+
+### Usage Examples
+
+Using the new Type Definition is possible to achieve the following:
+
+**groups that contain less than 10 `Post` written in 2022**
+
+```graphql
+query {
+  groups(where: {
+    postsAggregate: {
+      queryWhere: {
+        released: 2022
+      }
+      aggregationWhere: {
+        count_LT: 10
+      }
+    }
+  }) {
+    name
+  }
+}
+```
+
+**Cypher**
+```cypher
+MATCH (this:`Group`)
+CALL {
+    WITH this
+    MATCH (this:`Group`)-[this0:HAS_POST]->(this1:`Post`)
+    WHERE this1.released = 2022
+    RETURN count(this1) < $param0 AS var2
+}
+WITH *
+WHERE var2 = true
+RETURN this { .name } AS this
+```
+
+**groups that contain less than 10 `Post` written in 2022 or 2020**
+
+```graphql
+query {
+  groups(where: {
+    postsAggregate: {
+      queryWhere: {
+        OR: [
+            {
+                released: 2022
+            },
+            {
+                released: 2020
+            },
+        ]
+      }
+      aggregationWhere: {
+        count_LT: 10
+      }
+    }
+  }) {
+    name
+  }
+}
+```
+
+**Cypher**
+```cypher
+MATCH (this:`Group`)
+CALL {
+    WITH this
+    MATCH (this:`Group`)-[this0:HAS_POST]->(this1:`Post`)
+    WHERE this1.released = 2022 OR this1.released = 2020
+    RETURN count(this1) < $param0 AS var2
+}
+WITH *
+WHERE var2 = true
+RETURN this { .name } AS this
+```
+
+**groups that contain less than 10 `Post` written in 2022 or 2020, or with a title that is longer than 10 characters**
+
+```graphql
+query {
+  groups(where: {
+    postsAggregate: {
+      queryWhere: {
+        OR: [
+            {
+                released: 2022
+            },
+            {
+                released: 2020
+            },
+        ]
+      }
+      aggregationWhere: {
+        OR: 
+        [
+            {
+                count_LT: 10
+            },
+            {
+                count_LT: 10
```

Comment:
Is one of these conditions a typo. They don't seem to line up with the title/cypher so I'm not too sure what this example is trying to show?

Reply:
Good catch!